spec/models/manageiq/providers/kubernetes/container_manager_spec.rb

```diff
@@ -47,4 +47,18 @@
       expect(job.zone).to eq("default")
     end
   end
+
+  context "Kubeclient" do
+    it "receives expected arguments" do
+      require 'kubeclient'
+      expect(
+        Kubeclient::Client.new(
+          URI::HTTPS.build(:host => 'hostname', :port => 345),
+          'v1',
+          :ssl_options    => { :verify_ssl => OpenSSL::SSL::VERIFY_NONE },
+          :auth_options   => { :bearer_token => "1234" },
+          :http_proxy_uri => VMDB::Util.http_proxy_uri.to_s
+        )).to be_truthy
+    end
+  end
```

Comment:
This is getting towards the right idea, but isn't testing anything from the container manager.

It looks like the public container manager method on each provider is `raw_connect` or `<service>_connect`. You should be testing one of these to ensure that when you call it, `Kubeclient::Client` is receiving the arguments you expect it to.

Something along these lines:

```ruby
describe ".raw_connect" do
  let(:hostname) { "something"}
  let(:port)          { "1234"}
  let(:options      { { :some => :stuff } }

  it "uses the proxy if one is set" do
    allow(VMDB::Util).to receive(:http_proxy_uri).and_return("some.proxy")
    expect(Kubeclient::Client).to receive(:new).with(instance_of(URI::HTTPS, 'v1', hash_including(:http_proxy_uri => "some.proxy"))
    described_class.raw_connect(hostname, port, options)
  end
end
```